### DISCUSSION TITLE
How to detect current launcher and event, when it's changed?

### DISCUSSION THREAD

**Discussion by d-mozulyov** [microsoft/vscode-discussions#431] [Extension Development QnA] [answered]
While the extension is running, I need to know which launcher is the current one. In addition, I need to keep track of when the current launcher changes.

How to do it?

![image-2](https://user-images.githubusercontent.com/10961593/215952377-e83fb16e-73c6-448a-9640-5618be07c20a.png)

**Comment by isidorn** [comment]
This is currently not possible using the Extension API.

@roblourens might have more details if we plan to add something like this.

**Comment by d-mozulyov** [comment, marked as answer]
Can I ask someone to add this to the API? A very useful feature.

**Reply by r3m0t** [reply]
You can open an issue on the microsoft/vscode repository, it would help to include the information of how your extension will use this information. Do you need an API or would a context key for `viewsWelcome.when` be enough?

**Reply by d-mozulyov** [reply]
Thank you very much for drawing attention to my problem. I have detailed the context here: https://github.com/microsoft/vscode/issues/173788
